rcl/src/rcl/logging_rosout.c

```diff
@@ -173,11 +173,15 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(
   const rosidl_message_type_support_t * type_support =
     rosidl_typesupport_c__get_message_type_support_handle__rcl_interfaces__msg__Log();
   rcl_publisher_options_t options = rcl_publisher_get_default_options();
-  // Late joining subscriptions get the last 10 seconds of logs, up to 1000 logs.
-  options.qos.depth = 1000;
-  options.qos.durability = RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL;
-  options.qos.lifespan.sec = 10;
-  options.qos.lifespan.nsec = 0;
+
+  // Late joining subscriptions get the user's setting of rosout qos options.
+  const rcl_node_options_t *node_options = rcl_node_get_options(node);
+  if (NULL == node_options){
```

Comment:
using https://github.com/ros2/rcl/blob/70d13257a97f1278fd8e8a38f380e33b011f32e7/rcl/include/rcl/error_handling.h#L33 is a bit easier

Reply:
Ok, got it.